roles/splunk_deployer/tasks/main.yml

```diff
@@ -67,4 +67,8 @@
 - include_tasks: ../../../roles/splunk_common/tasks/disable_installed_apps.yml
   when: "'apps_location' in splunk and splunk.apps_location"
 
+# Now install local apps in apps_location_local exists
+- include_tasks: ../../../roles/splunk_common/tasks/provision_apps_local.yml
```

Comment:
Seeing that the disable_installed_apps.yml is called before this, does this guarantee that a local app is still enabled? Let's say the deployer happens to restart which will trigger the playbooks to run again. I'm not sure if we're preserving idempotency and as a result, the local app will end up disabled after the reboot.

Reply:
I'll try this out with a deployer reset.  It is difficult to preserve the existing functionality of `apps_location` and add a local app install due to the fact that to deploy apps to the Cluster/SH bundle they are installed locally then copied to the bundle and disabled.  So this can lead to a case where, for example, let's say we have app1 and app2 for the SHC and app3 local.  What will happen is:

1. app1 and app2 are installed locally
2. locally installed apps are copied to the sh_bundle (including app1 and app2)
3. app1 and app2 are disabled
4. app3 is installed locally.

This works fine the first time around.  If we then run the playbook again what happens is:
1. app1 and app2 are installed locally
2.  locally installed apps are copied to the sh_bundle (including app1, app2, and app3)
3. app1, app2, and app3 are disabled locally
4. app3 is installed/re-enabled locally

Now app3 has migrated to the cluster.  This will actually happen with any locally installed/enabled app, installed through ansible or manually, when the playbook is run.  With the current method of installing all apps local and copying them to the bundle, I'm not sure there is a way around this.  We would have to change the way apps are unpacked/installed for the bundle which would likely effect the way premium apps such as ITSI and ES are installed and be a much larger diff.

Reply:
Yeah, that's the concern I had as well. The 2nd scenario is what I expect you'll hit with the operator/kubernetes, given the long-lived-ness of Splunk deployments and the possibility of pod restarts that can occur during the lifecycle of a each container. 

I heard there's some app management ERD/RFC around the operator, I'm not sure if this work is part of that initiative or if this is just a temporary workaround while those details are being refined. But I'm sure you're going to subject to the limitations of case 2 with this change.

Reply:
Thanks Nelson!  Yes this `scope` (i.e. cluster or locally installed apps) will be used as part of the operators app installation framework.  For that work this fix would be used to install for the app installation when the Splunk containers are first brought up.  For "on the fly" update/install of apps, this would be an issue.